tests/settings/test_settings.py

```diff
@@ -18,3 +18,28 @@ def test_invalid_storage() -> None:
         assert False, f"'validate_settings' raised an exception {exc}"
     finally:
         cluster[0]["storage_sets"].remove("non_existing_storage")
+
+
+def test_topics_sync() -> None:
+    from copy import deepcopy
+
+    from snuba.settings.validation import validate_settings
+    from snuba.utils.streams.topics import Topic
+
+    all_settings = {
+        key: value
+        for key, value in settings.__dict__.items()
+        if not key.startswith("__") and not callable(key)
+    }
+    topic_map = all_settings["KAFKA_TOPIC_MAP"]
+    default_map = deepcopy(topic_map)
+    # Remove default settings and build dummy Topic Map
+    topic_map.clear()
+    topic_map = {t.value: {} for t in Topic}
+
+    try:
+        validate_settings(all_settings)
+    except Exception as exc:
+        assert False, f"'validate_settings' raised an exception {exc}"
```

Comment:
Is this `assert False` necessary?

Reply:
Could you please provide a more specific error message (this is what people will see when the test fails). The message should tell us that the test failed because the list of Topics defined does not correspond to the list in the validator.

Reply:
I would just not bother catching the exception, unless it's more specific than `Exception`, since the error will make the test fail

Reply:
I think it would be good to modify the error handling here to be specific to this test case, which is to check whether the two Topics lists correspond to one another. Changing it now.

Reply:
Also, I think the `assert False` is necessary to actually show that the test has failed when running pytest

Reply:
In that case, define a validation exception and raise that

Reply:
Sounds good. In that case should the above previous test also be modified in that way?

Reply:
I don't actually agree that `assert False` is necessary to show a failure in pytest:

```
➜  snuba git:(load-entity-from-yaml) ✗ cat tests/test_failure.py 
def test_failure():
	raise NotImplementedError() 

def test_passes():
	assert 1 + 1 == 2
➜  snuba git:(load-entity-from-yaml) ✗ pytest -vv tests/test_failure.py
=================================================================================================================================================================== test session starts ===================================================================================================================================================================
platform darwin -- Python 3.8.12, pytest-7.1.2, pluggy-0.13.0 -- /Users/oliver/workplace/snuba/.venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/oliver/workplace/snuba, configfile: setup.cfg
plugins: cov-3.0.0
collected 2 items                                                                                                                                                                                                                                                                                                                                         

tests/test_failure.py::test_failure FAILED                                                                                                                                                                                                                                                                                                          [ 50%]
tests/test_failure.py::test_passes PASSED                                                                                                                                                                                                                                                                                                           [100%]

======================================================================================================================================================================== FAILURES =========================================================================================================================================================================
______________________________________________________________________________________________________________________________________________________________________ test_failure _______________________________________________________________________________________________________________________________________________________________________
Traceback (most recent call last):
  File "/Users/oliver/workplace/snuba/tests/test_failure.py", line 2, in test_failure
    raise NotImplementedError()
NotImplementedError
================================================================================================================================================================= short test summary info =================================================================================================================================================================
FAILED tests/test_failure.py::test_failure - NotImplementedError
=============================================================================================================================================================== 1 failed, 1 passed in 7.07s ===============================================================================================================================================================
```

I think we should avoid catching an exception unless we know how to handle it/are doing something specific with it. I agree we can rewrite the above test to just not have a try/catch

Reply:
The new test does not have an `assert False` (although the test above it, test_invalid_storage, does). Were you referring to removing the `assert` in that test?

I'm not entirely sure how catching the exception can be avoided in this case. If I understood the JIRA task correctly, we are supposed to temporarily modify KAFKA TOPIC MAP and run `validate_settings` on this new settings. `validate_settings` will throw an exception, see here: https://github.com/getsentry/snuba/blob/76b9d912e2ca4e1d5579a698d07450ba7c7ac43d/snuba/settings/validation.py#L62-L64. Since I think we are supposed to use this function, and this function throws an exception when the lists are not in sync (which is what we're checking for), I'm not sure how else we can make sure the test fails when it should. Unless we don't have to use `validate_settings`